tests/unit/core/test_generate.py

```diff
@@ -86,9 +86,11 @@ def test_generate_single_error(dotsbws_error_result, caplog, parser):
     for record in caplog.records:
         if 'Keeping 0/1 read lines from {}'.format(dd) in record.getMessage():
             break
-    else:
```

Comment:
The else clause of a for loop is only executed if the for loop didn't break. 

If the `break` is hit, everything is fine. 

If the `break` is not hit, the else clause is executed and we `assert None` with the error message 'Unable to find log line indicating 0 success results in data file'.